crates/ra_syntax/src/ast/mod.rs

```diff
@@ -65,6 +65,24 @@ pub trait AttrsOwner<'a>: AstNode<'a> {
     }
 }
 
+pub trait DocCommentsOwner<'a>: AstNode<'a> {
+    fn doc_comments(self) -> AstChildren<'a, Comment<'a>> { children(self) }
+
+    /// Returns the textual content of a doc comment block as a single string.
+    /// That is, strips leading `///` and joins lines
+    fn doc_comment_text(self) -> String {
+        self.doc_comments()
+            .map(|comment| {
+                let prefix = comment.prefix();
+                let trimmed = comment.text().as_str()
+                    .trim()
+                    .trim_start_matches(prefix)
+                    .trim_start();
+                trimmed.to_owned()
```

Comment:
Do we really need this `to_owned` call? It's ok if we need it, but perhaps we can get rid of it?

Reply:
I needed it because `trimmed` was outliving `comment` and I was fighting the borrow checker. I will see if I can improve it tomorrow.

Reply:
Ok I can't seem to get rid of it but my rustfu is not that strong.